eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -603,21 +610,23 @@ def rule(self, matches):
         reversible = False
         if arrow == "<=>":
             reversible = True
+        rate1 = Rate(rate1) if rate1 else None
+        rate2 = Rate(rate2) if rate2 else rate1
```

Comment:
Does this still make sense if there can be only 0 or 2 rates?

Reply:
That's true. It does look misleading in the context of having either 0 or 2 rates. But based on how the grammar is defined, it should not cause any issue because either both `rate1` and `rate2` are defined, or `rate2` is `None` but in that case also `rate1` is `None`.

To address this, I've changed the implementation to more closely resemble its previous version in commit dffaea0, which I hope makes it less confusing.

Reply:
Indeed, the functionality should be the same because the grammar would not allow you to write such a case. But the implementation is now more straightforward.